cmd/draft/init.go

```diff
@@ -194,6 +219,91 @@ func (i *initCmd) setupDraftd() error {
 	return nil
 }
 
+// setupDriverEnv prepares os.Env for the driver. It operates on os.Env because
+// the plugin subsystem itself needs access to the environment variables
+// created here. This extends setupPluginEnv by giving init drivers additional
+// context on what feature flags were set.
+func (i *initCmd) setupDriverEnv() {
+	envMap := map[string]string{
+		"DRAFT_INIT_DRIVER_NAME":     i.driver,
+		"DRAFT_INIT_DRAFT_NAMESPACE": draftNamespace,
+	}
+
+	if i.ingressEnabled {
+		envMap["DRAFT_INIT_INGRESS_ENABLED"] = "1"
+	}
+
+	if i.upgrade {
+		envMap["DRAFT_INIT_UPGRADE"] = "1"
+	}
+
+	for key, val := range envMap {
+		os.Setenv(key, val)
+	}
+}
+
+func (i *initCmd) setupDraftConfig() error {
+	i.draftConfig = &installerConfig.DraftConfig{
+		Image: i.image,
+	}
+	if err := i.tlsOptions(i.draftConfig); err != nil {
+		return err
+	}
+	log.Debugf("raw chart config: %s", i.draftConfig)
+	return nil
+}
+
+func (i *initCmd) prepareDraftChart() error {
+	chartFiles := installer.DefaultChartFiles
+	if tlsEnable {
+		chartFiles = append(chartFiles, &chartutil.BufferedFile{
+			// TODO: Add chartutil.SecretName to k8s.io/helm/pkg/chartutil/create.go
+			Name: path.Join(chartutil.TemplatesDir, "secret.yaml"),
+			Data: []byte(installer.DraftSecret),
+		})
+	}
+	valuesFile := chartutil.BufferedFile{
+		Name: chartutil.ValuesfileName,
+		Data: []byte(i.draftConfig.String()),
+	}
+	chartFiles = append(chartFiles, &valuesFile)
+
+	chart, err := chartutil.LoadFiles(chartFiles)
+	if err != nil {
+		return err
+	}
+
+	// identical chart config should end up with the same chart
+	key, err := cache.Key(fmt.Sprintf("%x", md5.Sum([]byte(i.draftConfig.String()))))
+	if err != nil {
+		return err
+	}
+
+	cachePath := i.home.Path("cache", "charts", key)
```

Comment:
Hey not sure I'm following here. What are we intending to save under `$DRAFT_HOME/cache/charts` here?

Reply:
on line 287, we call `chartutil.Save` to write the Draft chart to disk. That path is then sent to the plugin as its first argument so it knows which chart to install. :)